docs: Update multi-hop concept doc (#4552)
hashicorp/boundary: website/content/docs/concepts/connection-workflows/multi-hop.mdx

Multi-hop sessions let you connect to private endpoints without necessarily opening inbound access to isolated or private networks.
By allowing outbound access from an egress worker in the private network to an upstream worker on a specific port, a Boundary client can access the target using a secure tunnel that is initiated by an ingress worker in the chain.

## Introduction

Multi-hop sessions introduces five concepts for how workers function.
You can categorize them from the point of view of a client and a worker.

From a worker perspective, the following concepts apply:

- Upstream worker: A worker that is closer in hop count to a controller.
Upstream workers receive join requests from downstream workers.
- Downstream worker: A worker that is closer in hop count to a target.
Downstream workers initiate connections to upstream workers.

From a client perspective, the following concepts apply:

- Ingress worker: A worker that Boundary clients connect to.
It is the first worker in the multi-hop chain.
- Intermediate worker: A worker with upstream and downstream workers in the chain.
These workers provide the fabric for multi-hop from ingress worker to egress worker.

> - Egress worker: The last worker int he multi-hop chain.

Reviewer: ```suggestion
- Egress worker: The last worker in the multi-hop chain.
```
Fixing typo

Boundary client connections egress the proxy chain at this worker to reach the target.

Note that ingress, intermediary, and egress are general ways to describe how the respective worker interfaces with resources.

> A worker can act as more than one of those at a time.

Reviewer: ```suggestion
A worker can act as more than one of these at a time.
```

Think this sounds better as `these` instead of `those`

Author: Yep, agreed. Thank you!

> ## Multi-hop worker capabilities

Reviewer: Having read the preview, I'm thinking we move this section to be under the `Introduction` section, and we move the content from the `Introduction` section here and rename the header from `Multi-hop worker capabilities` to `Multi-hop worker concepts`.  We talk about things upstream and downstream here currently, and if we move that to be under the introduction, then that is a good segway to better explain the concepts so users understand what was just mentioned in more detail.

Author: > Having read the preview, I'm thinking we move this section to be under the `Introduction` section, and we move the content from the `Introduction` section here and rename the header from `Multi-hop worker capabilities` to `Multi-hop worker concepts`. We talk about things upstream and downstream here currently, and if we move that to be under the introduction, then that is a good segway to better explain the concepts so users understand what was just mentioned in more detail.

Yep, great suggestion! I just moved these around and I think it flows nicely. I think this topic is much improved :tada:

I am wondering if we should expand on why people would want to configure multi-hop? We briefly address the benefits in the first para, but I'm wondering if it could be stronger.

Reviewer: I reread the intro and thought about it some more, and honestly, Im coming up without any improvements.  The value of multi-hop workers seems clear to me with that intro, but I also have been working with multi-hop workers a lot so perhaps another set of eyes on it would be good?

Author: Sounds good! I would like to bring in some other reviewers. Thank you for all your help @sdoyle88 . Robin and I still need to review your updates to the worker procedure too. We'll do those things and get back to you.
